database/engine/metadata_log/metalogpluginsd.c

```diff
@@ -4,6 +4,63 @@
 #include "metadatalog.h"
 #include "metalogpluginsd.h"
 
+PARSER_RC metalog_pluginsd_host_action(
+    void *user, char *machine_guid, char *hostname, char *registry_hostname, int update_every, char *os, char *timezone,
+    char *tags)
+{
+    struct metalog_pluginsd_state *state = ((PARSER_USER_OBJECT *)user)->private;
+
+    RRDHOST *host = rrdhost_find_by_guid(machine_guid, 0);
+    if (host)
+        goto write_replay;
+
+    if (strcmp(machine_guid, registry_get_this_machine_guid()) == 0) {
```

Comment:
The `registry_get_this_machine_guid()` function opens or creates a file under `var/lib/netdata`, and at this point I think we don't want to do that.

Reply:
It has already been called at this point (before the metalog_init)